src/gt4py/next/iterator/type_system/inference.py

```diff
@@ -96,6 +96,17 @@ def _set_node_type(node: itir.Node, type_: ts.TypeSpec) -> None:
     node.type = type_
 
 
+def copy_type(from_: itir.Node, to: itir.Node) -> itir.Node:
```

Comment:
I personally find a bit confusing the name of the function given its signature returning the updated node. I would suggest to either rename the function (e.g. `update_node_type`?) or keep the current name and just make it return `None`

Reply:
I agree, I picked the latter.